pandas/core/computation/eval.py

```diff
@@ -232,8 +237,23 @@ def eval(expr, parser='pandas', engine=None, truediv=True,
     pandas.DataFrame.query
     pandas.DataFrame.eval
     """
-    inplace = validate_bool_kwarg(inplace, 'inplace')
-    first_expr = True
+
+    def is_modifiable(obj):
+        target_type = type(obj).__name__
+        return hasattr(obj, "__dict__") or target_type == "dict"
+
+    inplace = validate_bool_kwarg(inplace, "inplace")
+
+    # "Primitives" cannot be modified inplace because
+    # they have no attributes that we can modify.
+    #
+    # The only exceptions are lists and dictionaries,
```

Comment:
I don't think it is possible to modify lists -> `__getitem__` is used, and it is a string expression, so you can never work with a list

Or do you have an example where this actually works?

(anyway, it would also be good to add some tests for those statements, also for the dictionaries)